### PR TITLE
add coveralls auto tool to json-c

### PR DESCRIPTION
**Pull request by dota17**
This PR adds the coveralls auto tool to json-c.  [Coveralls](https://coveralls.io/) is a web service to help you track your code coverage over time, and ensure that all your new code is fully covered.

I have added it in [dota17/json-c](https://github.com/dota17/json-c?organization=dota17&organization=dota17) and showed a PR submission example in [here](https://github.com/dota17/json-c/pull/40). We can see the json-c's  code coverage in README.md and get the code coverage report from coveralls automatically when you submit a PR. It is a useful and helpful tool.

Following are the configuration steps：
1. Sign in the https://coveralls.io/ with github account and open the switch of json-c/json-c in [```ADD REPO``` ](https://coveralls.io/repos/new).
2. configure file: 

note: ```Compact``` means that you will get a neat and clear report. 
3. Merge this pr.
4. Get and add the badge into README.md 

These steps are simple and coveralls is an interesting and useful tool.  You will like it. （‐＾▽＾‐）

### CODE REVIEW

**Comment by dota17** [comment]
And, we should choose suitable files to report. In [here](https://github.com/dota17/json-c/blob/c684b1d40bb22cf214e03d83e63e149a2499653f/.travis.yml#L96), I have chosen the most of the .c files and partial .h files.

File: CMakeLists.txt [hunk 1]
```diff
@@ -197,6 +197,7 @@ message(STATUS "Written ${PROJECT_BINARY_DIR}/json_config.h")
 
 if ("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
     set(CMAKE_C_FLAGS           "${CMAKE_C_FLAGS} -ffunction-sections -fdata-sections")
+    set(CMAKE_C_FLAGS           "${CMAKE_C_FLAGS} -fprofile-arcs -ftest-coverage")
```
Comment (hawicz) [inline CMakeLists.txt]: This is going to add overhead and change the behavior (i.e. result in .gcda files being emitted) and thus can't be enabled in a regular build.

Comment (dota17) [reply CMakeLists.txt]: You are right, it's not suitable to add "-fprofile-arcs -ftest-coverage" in this regular build. I will try to make it in a new way.

File: .travis.yml [hunk 2]
```diff
@@ -78,6 +78,23 @@ matrix:
...
+        - sudo pip install cpp-coveralls
+      script:
+        - mkdir build
+        - cd build
+        - cmake ..
+        - make
+        - make test
+        - cd -
+      after_success:
+        - coveralls --include arraylist.c --include arraylist.h --include json_object.c --include json_object.h --include json_object_iterator.c --include json_object_iterator.h --include json_object_private.h --include json_pointer.c --include json_pointer.h --include json_tokener.c --include json_tokener.h --include json_util.c --include json_util.h --include json_visit.c --include json_visit.h --include linkhash.c --include linkhash.h --include printbuf.c --include printbuf.h --include random_seed.c --include strerror_override.c
```
[the remark this answers is not in the document]
Comment (dota17) [reply .travis.yml]: For getting the report of suitable documents selectively, I choose to```--include``` every document inconveniently. I will use the wildcard and see how it works.

[the remark this answers is not in the document]
Comment (dota17) [reply .travis.yml]: ```--include``` does not support the wildcard and I choose the ```--exclude```. So the report contains some meaningless files,such as debug.c, json_c_version.c and so on. But this result is acceptable.

**Comment by dota17** [comment]
I have modified the ```.travis.yml```.  And I think it is necessary and worthy to run it in a separate service. It won't affect other services.

**Comment by dota17** [comment]
Because json-c/json-c repository does not have the usage rights of [coveralls](https://coveralls.io/), this PR could not pass. I suggest you  give the permission to Coveralls as the steps shown in PR description.（‐＾▽＾‐）

**Comment by dota17** [comment]
@hawicz  I have modified the ```.travis.yml```.  The CI couldn't pass and maybe need you to give the permission to Coveralls.   Coveralls has provided a lot of convenience for community submission and code testing. It really is a good tool, ≧◉◡◉≦

**Comment by dota17** [comment]
@hawicz  I have opened the permissions of ```Coveralls``` and the CI has passed. Do you have any other opinions on this？

**Comment by dota17** [comment]
This PR is not associated with the core code, so I will merge it. We can try it out. If it is not suitable, we can revise/delete it.

**Pull request merged**